internal/updater/verify.go

```diff
@@ -0,0 +1,281 @@
+1SP8nCgw6qTtWX/SWOMLbWUHIvqIRgQQEQIABgUCSMWDIQAKCRCs2KSKiKRmEaIU
+AKCoPxyFiqvSggiYy/QhzRhOcR3iHgCePZfdR1lruR4YLKaU3+jAX0uxPHCIRgQQ
+EQIABgUCSMmMgwAKCRATnMhm/FBasKoHAKC2xrIKwj4cS+eYFVKewmhJu1/xeQCg
+rvN48fHB0l9MTX2buqtajgqBX4GIRgQQEQIABgUCSM4OMgAKCRBI2RcA+sN2ZZsP
+AKDLyUJKz3WONgPY3Sgzg4PhG/yeEgCfaw9J/XRU7rgj3MYs8Cm29uaCW3eIRgQQ
+EQIABgUCSM6bagAKCRDDGy9Vk+Kvk+TrAJ9Ykff3BwvD8PyjwEJvWcHgxxLhYwCf
+UXMdb6mC6yAdU9IpFpmLXD7KdoGIRgQQEQIABgUCSUvV3AAKCRDnUuMnWr3CRszu
+AJ9opkxi6TuwxItNRKear3zKtgEBmwCcDKXh3b1eLssnQu+3vMcIzGkdoTqIRgQQ
+EQIABgUCSZhnnAAKCRA8TejA5LSMSMA5AJ43j3a/f0oEU2i+aJGQ4DfvK3A7PwCd
+HvA8ozLsdQuSifhiyWPnSfiKi+2JARwEEAECAAYFAksnlSkACgkQ1Iu7irZrI6Cd
+hAgAoQjTNcDxv7fcA/nuS15/3Q2PoGBW8rX/iMobbxTvpwQh0t3ICzCEcTE2UfjY
+/+fRZHrmgINcQxaw0uqResOUKjqu6nzpv4U4gEttZAilmqOJDtoDvjHx30cS8pRH
+vs/uH1xUuwpj8y0IXStT8jzau22H1mSSitjD8VUnRVJqFP15HO5uQQrnZmGA2GcQ
+110knob3ZpNLir1aKpvV99dRiVURPA6Hs9aOfiujRSIERon2b2GphU0q194E3KYF
+mwmMupnQJo2k3jKcULsq6zJL286AHA2eVKORpPoqPIrKDQQpcXt+1bfnq1xTvgO6
+34HAQSlYj9nq1IrMZTNA28HEwrQgRG9taW5payBTY2h1bHogPG1haWxAZHNjaHVs
+ei5ldT6IYgQTEQIAIgUCUzHnoAIbAwYLCQgHAwIGFQgCCQoLBBYCAwECHgECF4AA
+CgkQ0x+gVMha76wQ8ACfXPu8TJLjHco89g1S0uup6d8GXNsAoIIbJqoYpSE+0Q5V
+8DE1U4x20y0+uQINBEiykPYQCACPqgDb8Er5uDMcYYRr7g4wUDjgSE7AG9FYB6Tz
+7/6ZjRAZk0xuCbzQhfJFBya7jHUhUAFMC0zNCglL8r21s8Ocy10L/HSMKXozVkvn
+nLvBGbgMSNxMGX4OPaph1OV+7mrrW1Ogxv9YOkk9mEoKxUOFRQ7wDzaqzbzJRRgu
+ZhDM1XHM/BUOW23gm7dotplKf/fvsRJtsIDDPssrmKq94aQxXc1kJ5S23TMVnky3
+hCr+TJccK2PKX4J8cLiF9m33fepzDqMrShOP2Hmlrv4RwDDgjiZoiQtmFgzD0qvO
+ft4EaVDkoyeGPYuwKyyQ0njdG68HGiw+5OxFXAMwY09uKRJHAAMFCACMLt5HJlmh
+z0MFuaztihfQGgZ8AkcWROiEHXv4/D3M+aYNcdr51CQLquDgC9NgBNu7oV4yfXSO
+3b6QgZBuXLCCPN+KIXoF9ULD+r1lCwLZTV7o4gXCOX2/DP01hlcz7WJAZQrYY6R/
+DCLWhGT1jQBas/MHfD0CjHagj/BpHuGbUfgTYcQRkPbLcqUYClUMYfldVuHquwZQ
+Q4hgGiMIJMPbW1CXPzHBXZ+lDpMyTuJZ77uXipLsM8G9zi9KV6NhqgocYZX6GB5Y
+AvUWNqTAqwACCvi60RPnU8a5G+GGIy53ymJWITCdECar5JceJ+dH3HCiP7ElMsD9
+bTZ+YLlgTZi2iEkEGBEKAAkCGwwFAlZOuaoACgkQ0x+gVMha76zV1QCgjgtcqRrB
+FLubt+/9smrMOkp0TBAAnjvaL3UpZGHeYWoD+Hkob5DVOugQ
+=EERM
+-----END PGP PUBLIC KEY BLOCK-----
+`)
+
+func gpgVerify(data, sig []byte) (bool, error) {
+	keyring, err := openpgp.ReadArmoredKeyRing(bytes.NewReader(pubkey))
+	if err != nil {
+		debug.Log("failed to read public key: %q", err)
+		return false, err
+	}
+
+	_, err = openpgp.CheckArmoredDetachedSignature(keyring, bytes.NewReader(data), bytes.NewReader(sig))
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// retrieve the hash for the given filename from a checksum file
+func findHashForFile(buf []byte, filename string) ([]byte, error) {
+	s := bufio.NewScanner(bytes.NewReader(buf))
+	for s.Scan() {
+		p := strings.Split(s.Text(), "  ")
+		if len(p) < 2 {
+			continue
+		}
+		if p[1] != filename {
+			continue
+		}
+		h, err := hex.DecodeString(p[0])
+		if err != nil {
+			return nil, err
+		}
+		return h, nil
+	}
+
+	return nil, fmt.Errorf("hash for file %q not found", filename)
```

Comment:
Ideally the hashsum should be recomputed and compared to that in the hashsum file

Reply:
Yes, of course. This wasn't ready for review, yet. Sorry.